api/resource_organizations.go

```diff
@@ -14,10 +15,22 @@ const (
 	OrganizationTypeShared   OrganizationType = "SHARED"
 )
 
-func (client *Client) GetOrganizations(ctx context.Context) ([]Organization, error) {
+type organizationFilter struct {
+	admin bool
+}
+
+func (f *organizationFilter) apply(req *graphql.Request) {
+	req.Var("admin", f.admin)
+}
+
+type OrganizationFilter func(*organizationFilter)
+
+var AdminOnly OrganizationFilter = func(f *organizationFilter) { f.admin = true }
```

Comment:
my 2c, I find this to be an overcomplicated way of writing:

```go
func AdminOnly(f *organizationFilter) {
  f.admin = true
}
```

I understand you want to highlight AdminOnly fits OrganizationFilter type, but that is going to be catch by the compiler anyways, isn't it?

Reply:
Yeah. It's mostly for docs:

<img width="857" alt="image" src="https://user-images.githubusercontent.com/1144197/224394705-6e238603-75e0-4e20-ba04-9d15c6715024.png">

I could go either way.

Reply:
fair point :) TIL